jax/_src/scipy/linalg.py

```diff
@@ -2243,3 +2243,45 @@ def _binom(n, k):
   b = lax.lgamma(n - k + 1.0)
   c = lax.lgamma(k + 1.0)
   return lax.exp(a - b - c)
+
+
+@jit
+def solve_sylvester(A: jnp.ndarray, B: jnp.ndarray, C: jnp.ndarray) -> jnp.ndarray:
```

Comment:
Input arrays should be annotated with `ArrayLike`, and output should be annotated with `Array`, similar to other APIs in this file.

With this change, you'll have to preprocess the inputs (using e.g. `promote_args_inexact`) to convert them to JAX arrays using `jnp.asarray`, or else you'll get type checking errors because `ArrayLike` doesn't support the `.shape` attribute